service/service.go

```diff
@@ -57,28 +48,29 @@ type serviceState struct {
 // ServiceClient is an interface for interacting with the Service actor.
 // It is created when the Service is started.
 type ServiceClient struct {
-	serviceStarted    *atomic.Bool
+	start             chan struct{}
```

Comment:
not clear to me the diff btwn this `start` and the one in `Service`

Reply:
Refactored and removed `ServiceClient`.